lib/teleterm/vnet/service.go

```diff
@@ -18,31 +18,258 @@ package service
 
 import (
 	"context"
-	"math/rand"
-	"time"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"sync"
 
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/types"
 	api "github.com/gravitational/teleport/gen/proto/go/teleport/lib/teleterm/vnet/v1"
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/teleterm/api/uri"
+	"github.com/gravitational/teleport/lib/teleterm/daemon"
+	logutils "github.com/gravitational/teleport/lib/utils/log"
+	"github.com/gravitational/teleport/lib/vnet"
 )
 
+var log = logutils.NewPackageLogger(teleport.ComponentKey, "term:vnet")
+
 // Service implements gRPC service for VNet.
 type Service struct {
 	api.UnimplementedVnetServiceServer
+
+	cfg       Config
+	mu        sync.Mutex
+	isRunning bool
+	isClosed  bool
```

Comment:
I'm wondering if these variables couldn't be merged into a single variable with more values.
Now you can theoretically have combinations that don't make sense, like `isRunning: true` and `isClosed: true`, no?

Reply:
That's true. I guess I could use an enum here for that. I don't think it's a popular practice in Go but it'd certainly work here.

Reply:
Thanks for changing this.